pkgconfig/CMakeLists.txt

```diff
@@ -41,21 +41,31 @@ list(JOIN ENCLAVE_CFLAGS_LIST " " ENCLAVE_CFLAGS_GCC)
 
 # See `cmake/apply_lvi_mitigation.cmake`.
 if (LVI_MITIGATION MATCHES ControlFlow)
-  set(LVI_MITIGATION_FLAGS -Wa,-mlfence-before-indirect-branch=register
-                           -Wa,-mlfence-before-ret=not)
-  if (NOT GLIBC_VERSION)
-    get_glibc_version()
-  endif ()
-  if (GLIBC_VERSION LESS 2.27)
-    set(LVI_MITIGATION_FLAGS ${LVI_MITIGATION_FLAGS} -g0)
+  if (OS_USE_CLANG_10)
+    set(LVI_MITIGATION_FLAGS -Wa,-mlfence-before-indirect-branch=register
+                             -Wa,-mlfence-before-ret=not)
+    if (NOT GLIBC_VERSION)
+      get_glibc_version()
+    endif ()
+    if (GLIBC_VERSION LESS 2.27)
+      set(LVI_MITIGATION_FLAGS ${LVI_MITIGATION_FLAGS} -g0)
+    endif ()
+    set(LVI_MITIGATION_FLAGS_GCC ${ENCLAVE_CFLAGS_GCC} ${LVI_MITIGATION_FLAGS}
+                                 -mindirect-branch-register)
+    list(JOIN LVI_MITIGATION_FLAGS_GCC " " ENCLAVE_LVI_MITIGATION_CFLAGS_GCC)
+
+    set(LVI_MITIGATION_FLAGS_CLANG ${ENCLAVE_CFLAGS_CLANG_LIST}
+                                   ${LVI_MITIGATION_FLAGS} -no-integrated-as)
+    list(JOIN LVI_MITIGATION_FLAGS_CLANG " "
+         ENCLAVE_LVI_MITIGATION_CFLAGS_CLANG)
+  else ()
+    # clang-11 LVI
+    set(LVI_MITIGATION_FLAGS -mlvi-cfi)
```

Comment:
Apologies if this is a silly question with an obvious answer, but why only control flow mitigations and not full LVI mitigations (https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mlvi-hardening)?

There also seems to be broader option that includes LVI CFI (https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mseses).

Reply:
Thanks for reviewing! This is not official answer but I think: 

1. why only control flow mitigations and not full LVI
Currently, OE only provides custom ControlFlow LVI mitigation through a custom toolchain, so this PR is only adding ControlFlow as well.

2. why not use `-mseses`
`-mseses` turns on speculative and LVI mitigation, where currently speculative load hardening for OE is on by default, but LVI is optional. LVI comes with performance impact, so it's an option for the customer to leverage.

Reply:
@achamayou 

`SESES` approach achieves broader protection by conservatively inserting LFENCEs, which is similar to the original full LVI mitigation that Intel introduces. At the time of bringing in the LVI mitigation to OE, we decided not to support that option because of its huge performance implications.

Reference: https://reviews.llvm.org/rGbf95cf4a6816476afd76d5b981ea5fdcb9771999#change-PwDumSOD4qd4

@asvrada 

Can you also test the performance impact of the `-mlvi-hardening`? Ideally, this option should be performant while providing better coverage against LVI.

Reference:
-mlvi-hardening: https://reviews.llvm.org/D75936
-mlvi-cfi: https://reviews.llvm.org/D75934

Reply:
@asvrada @mingweishih thank you for the clarification, that's very helpful.